wkcuber/downsampling.py

```diff
@@ -38,6 +38,12 @@ def determine_buffer_edge_len(dataset):
     return min(DEFAULT_EDGE_LEN, dataset.header.file_len * dataset.header.block_len)
 
 
+def calculate_scale(target_mag):
```

Comment:
Very elegant solution :+1: It didn't cross my mind that you could simply reuse `downsample_mags_anisotropic` by calculating an appropriate "fake scale".

However, I'd rename `calculate_scale` to `calculate_virtual_scale_for_target_mag` or something like that. Also you could add a comment that explains that this scale only serves the purpose of being used for `downsample_mags_anisotropic`. Otherwise, it could be confusing why the actual dataset scale is not used.